cmd/cli/docker/docker_run.go

```diff
@@ -0,0 +1,279 @@
+package docker
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"sigs.k8s.io/yaml"
+
+	"github.com/bacalhau-project/bacalhau/cmd/util/flags"
+	"github.com/bacalhau-project/bacalhau/cmd/util/handler"
+	"github.com/bacalhau-project/bacalhau/cmd/util/opts"
+	"github.com/bacalhau-project/bacalhau/cmd/util/parse"
+	"github.com/bacalhau-project/bacalhau/cmd/util/printer"
+	"github.com/bacalhau-project/bacalhau/pkg/bacerrors"
+	jobutils "github.com/bacalhau-project/bacalhau/pkg/job"
+	"github.com/bacalhau-project/bacalhau/pkg/model"
+	"github.com/bacalhau-project/bacalhau/pkg/system"
+	"github.com/bacalhau-project/bacalhau/pkg/util/templates"
+)
+
+var (
+	runLong = templates.LongDesc(i18n.T(`
+		Runs a job using the Docker executor on the node.
+		`))
+
+	//nolint:lll // Documentation
+	runExample = templates.Examples(i18n.T(`
+		# Run a Docker job, using the image 'dpokidov/imagemagick', with a CID mounted at /input_images and an output volume mounted at /outputs in the container. All flags after the '--' are passed directly into the container for execution.
+		bacalhau docker run \
+			-i src=ipfs://QmeZRGhe4PmjctYVSVHuEiA9oSXnqmYa4kQubSHgWbjv72,dst=/input_images \
+			dpokidov/imagemagick:7.1.0-47-ubuntu \
+			-- magick mogrify -resize 100x100 -quality 100 -path /outputs '/input_images/*.jpg'
+
+		# Dry Run: check the job specification before submitting it to the bacalhau network
+		bacalhau docker run --dry-run ubuntu echo hello
+
+		# Save the job specification to a YAML file
+		bacalhau docker run --dry-run ubuntu echo hello > job.yaml
+
+		# Specify an image tag (default is 'latest' - using a specific tag other than 'latest' is recommended for reproducibility)
+		bacalhau docker run ubuntu:bionic echo hello
+
+		# Specify an image digest
+		bacalhau docker run ubuntu@sha256:35b4f89ec2ee42e7e12db3d107fe6a487137650a2af379bbd49165a1494246ea echo hello
+		`))
+)
+
+// DockerRunOptions declares the arguments accepted by the `docker run` command
+type DockerRunOptions struct {
+	WorkingDirectory string // Working directory for docker
+
+	SpecSettings       flags.SpecFlagSettings       // Setting for top level job spec fields.
+	ResourceSettings   flags.ResourceUsageSettings  // Settings for the jobs resource requirements.
+	NetworkingSettings flags.NetworkingFlagSettings // Settings for the jobs networking.
+	DealSettings       flags.DealFlagSettings       // Settings for the jobs deal.
+	RunTimeSettings    flags.RunTimeSettings        // Settings for running the job.
+	DownloadSettings   flags.DownloaderSettings     // Settings for running Download.
+
+}
+
+const (
+	DefaultDockerRunWaitSeconds = 600
+)
+
+func NewDockerRunOptions() *DockerRunOptions {
+	return &DockerRunOptions{
+		SpecSettings: flags.SpecFlagSettings{
+			Verifier: model.VerifierNoop.String(),
+			// TODO most users would probably prefer IPFS
```

Comment:
since we are retuning default values, that are also expected to be common between docker and wasm (mostly), does it make sense to move this logic to `flags` package and have `flags.NewSpecFlagSettings()` return default settings? The same goes for the other settings

Reply:
Good idea, will do this for other common flags as well.